security/pfSense-pkg-tinc/files/usr/local/pkg/tinc.inc

```diff
@@ -0,0 +1,275 @@
+	if ($tincconf['tinc_down']) {
+		file_put_contents("{$configpath}/tinc-down", str_replace("\r", "", base64_decode($tincconf['tinc_down'])) . "\n");
+		chmod("{$configpath}/tinc-down", 0744);
+	}
+	if ($tincconf['host_up']) {
+		file_put_contents("{$configpath}/host-up", str_replace("\r", "", base64_decode($tincconf['host_up'])) . "\n");
+		chmod("{$configpath}/host-up", 0744);
+	}
+	if ($tincconf['host_down']) {
+		file_put_contents("{$configpath}/host-down", str_replace("\r", "", base64_decode($tincconf['host_down'])) . "\n");
+		chmod("{$configpath}/host-down", 0744);
+	}
+	if ($tincconf['subnet_up']) {
+		file_put_contents("{$configpath}/subnet-up", str_replace("\r", "", base64_decode($tincconf['subnet_up'])) . "\n");
+		chmod("{$configpath}/subnet-up", 0744);
+	}
+	if ($tincconf['subnet_down']) {
+		file_put_contents("{$configpath}/subnet-down", str_replace("\r", "", base64_decode($tincconf['subnet_down'])) . "\n");
+		chmod("{$configpath}/subnet-down", 0744);
+	}
+
+	$pfs_version = substr(trim(file_get_contents("/etc/version")), 0, 3);
+	if ($pfs_version == "2.2") {
+		$pbietcpath = '/usr/pbi/tinc-' . php_uname("m") . '/local/etc';
+		unlink_if_exists("{$pbietcpath}/tinc");
+		symlink($configpath, "{$pbietcpath}/tinc");
+	}
+
+	if ($tincconf['enable'] != "") {
+		tinc_write_rcfile();
+		if (is_service_running("tinc")) {
+			restart_service("tinc");
+		} else {
+			start_service("tinc");
+		}
+	} else {
+		if (is_process_running("tincd")) {
+			stop_service("tinc");
+		}
+		unlink_if_exists("/usr/local/etc/rc.d/tinc.sh");
+	}
+
+	rmdir_recursive("/usr/local/etc/tinc.old");
+	conf_mount_ro();
+}
+
+function tinc_write_rcfile() {
+	$rc['file'] = 'tinc.sh';
+	$rc['start'] .= "/usr/local/sbin/tincd --config=/usr/local/etc/tinc\n\t";
+	$rc['stop'] .= "/usr/local/sbin/tincd --kill \n\t";
+	write_rcfile($rc);
+}
+
+function tinc_install() {
+	global $config;
+
+	safe_mkdir("/usr/local/etc/tinc");
+	safe_mkdir("/usr/local/etc/tinc/hosts");
+	tinc_write_rcfile();
+	unlink_if_exists("/usr/local/etc/rc.d/tincd");
+	clear_log_file("/var/log/tinc.log");
+
+	/* Create Interface Group */
+	if (!is_array($config['ifgroups']['ifgroupentry'])) {
+		$config['ifgroups']['ifgroupentry'] = array();
+	}
+
+	$a_ifgroups = &$config['ifgroups']['ifgroupentry'];
+	$ifgroupentry = array();
+	$ifgroupentry['members'] = '';
+	$ifgroupentry['descr'] = 'tinc mesh VPN interface group';
+	$ifgroupentry['ifname'] = 'tinc';
+	$a_ifgroups[] = $ifgroupentry;
+
+	/* XXX: Do not remove this. WTH?! */
+	mwexec("/bin/rm -f /tmp/config.cache");
+
+	write_config("[tinc] Package installed.");
+}
+
+function tinc_deinstall() {
+	global $config;
+
+	/* Remove Interface Group */
+	if (!is_array($config['ifgroups']['ifgroupentry'])) {
+		$config['ifgroups']['ifgroupentry'] = array();
+	}
+
+	$a_ifgroups = &$config['ifgroups']['ifgroupentry'];
+
+	$myid = -1;
+	$i = 0;
+	foreach ($a_ifgroups as $ifgroupentry) {
+		if ($ifgroupentry['ifname'] == 'tinc') {
```

Comment:
Yes, this absolutely sounds feasible. Do you want me to do this change now or wait for upstream first to prevent users creating these names themselves?